osbs/build/build_request.py

```diff
@@ -867,7 +889,13 @@ def render(self, validate=True):
                            # add the path to the plugin's
                            # configuration. This is done elsewhere.
                            None):
-                          self.spec.registry_secrets.value})
+                          self.spec.registry_secrets.value,
+
+                          ('buildstep_plugins', 'orchestrate_build', 'osbs_client_config'):
```

Comment:
Ok, I think enabling the plugin to take a param for a custom location for osbs.conf is a good idea. It'll also help facilitate unit testing. 

My only concern is to have this being done by this PR. But it's not a major concern. I was hoping to fully configure that plugin in its own PR. Let's leave it if others don't have a problem with it.